src/plugins/intel_gpu/src/graph/program.cpp

```diff
@@ -631,7 +632,7 @@ void program::post_optimize_graph(bool is_internal) {
 // mark if the node is constant assuming that all dependencies are marked properly
 void program::mark_if_constant(program_node& node) {
     if (node.get_dependencies().empty() || node.is_type<prior_box>() ||
-        node.is_type<assign>() || node.is_type<read_value>()) {
+        node.is_type<assign>() || node.is_type<read_value>() || node.is_type<gather_nonzero>()) {
         return;
```

Comment:
nit: What do you think to assign false explicitly as below?

```suggestion
        node.is_type<assign>() || node.is_type<read_value>() || node.is_type<gather_nonzero>()) {
        node.constant = false;
        return;
```

Reply:
Looks good to me

Reply:
Because it is original behavior that only to mark "if constant" I think